layer/executables/layer_runtime.py

```diff
@@ -105,13 +108,17 @@ def _create_dataset(self, name: str, func: Callable[..., Any]) -> Any:
             ProjectId(value=str(self._project_id)), name, display_fabric
         )
         dataset = func()
+        dataset = check_and_convert_to_df(dataset)
```

Comment:
This check should probably not be there. Is it required for assertions? If it is just to ensure the `DataFrame` is passed here, I suggest not to solve this problem in this PR. This is the scope of https://linear.app/layer/issue/LAY-3464/add-validation-to-datasetmodel-decorators-to-check-function-signatures to solve the issue globally.

Reply:
Similar to the supported model flavors we need to make sure that we'd getting a dataframe from the function so that assertions could operator on that one.